### PR TITLE
Update library to target .NET Standard as well as .NET Framework

### PR DESCRIPTION
**Pull request by TAGC**
Updates this library so that the NewRelic sink can be used within applications running on .NET Core as well as .NET Framework.

This is a breaking change and should be signalled as such by bumping the major version number (1.0.3 -> 2.0.0).

The application name must be explicitly provided to the sink when calling `LoggerConfiguration.WriteTo.NewRelic(...)` as it's not possible to read this from `ConfigurationManager.AppSettings` (this is a .NET Framework-only feature). The README has been updated accordingly.

### CODE REVIEW

File: src/Serilog.Sinks.NewRelic/Serilog.Sinks.NewRelic.nuspec [hunk 1]
```diff
@@ -1,33 +0,0 @@
-﻿<?xml version="1.0" encoding="utf-8"?>
```
Comment (stanisls) [inline src/Serilog.Sinks.NewRelic/Serilog.Sinks.NewRelic.nuspec:1]: Why is nuspec removed? Shouldn't it be rather be updated accordingly?

Comment (TAGC) [reply src/Serilog.Sinks.NewRelic/Serilog.Sinks.NewRelic.nuspec:1]: With the update to using the newer csproj file format, a lot of the properties defined in the .nuspec file can go into the csproj file instead, and this is what [Microsoft recommends](https://docs.microsoft.com/en-us/nuget/reference/nuspec#project-type-compatibility):

> If you are creating a package using dotnet.exe pack or msbuild pack target, we recommend that you include all the properties that are usually in the .nuspec file in the project file instead. 

I think that .nuspec are still supported if you'd prefer that we keep it, though.

**Review by stanisls — COMMENTED** [review]
A good initiative.
I was actually considering to deprecate the whole library, as NewRelic Logs is the new and much-improved way to submit logging data to NR.
I see that the nuspec file was removed. If this is to be built and published at Nuget.org, I would require a nuspec file that reflect the new targets.
.Net Core uses JSON configuration of the application. Have you checked if NewRelic appName can be fetched from there and rewrite accordingly, instead of removing the default appSettings-based behaviour?

File: src/Serilog.Sinks.NewRelic/Serilog.Sinks.NewRelic.nuspec (hunk 1 above)
Comment (stanisls) [reply src/Serilog.Sinks.NewRelic/Serilog.Sinks.NewRelic.nuspec:1]: I see. We are still on .NET Framework 4.6 in the project which uses (or, rather, used) this library, so I haven't looked much into the finer points of the Core.

Comment (TAGC) [reply src/Serilog.Sinks.NewRelic/Serilog.Sinks.NewRelic.nuspec:1]: I've made some local changes (not yet pushed) to the csproj file (migrated over author and license information) as you can see in the screenshot below:

image

One thing though - it seems to include the icon when viewed through Package Explorer, but that icon isn't appearing in the NuGet Package Manager. I'm not sure if this is just a caching issue that will fix itself.

**Comment by TAGC** [comment]
> I was actually consider to deprecate the whole library, as NewRelic Logs is the new way and much-improved way to submit logging data to NR.

That's true, but unfortunately our company isn't licensed to use NewRelic Logs so we've had to fallback on feeding logs to NewRelic the old-fashioned way.

> .Net Core uses JSON configuration of the application. Have you checked if NewRelic appName can be fetched from there and rewrite accordingly, instead of removing the default assSettings-based behaviour?

ASP.NET Core prefers to go the dependency injection route as opposed to having things statically-accessible, so to access any configuration options you'd most likely need to pass an `IConfiguration` object as an argument to the sink setup method anyway.

Used alongside `Serilog.Settings.Configuration`, the approach we've taken is just to configure the following in our applications' `appSettings.json` file:

```jsonc
"Serilog": {
  "Using": [
    "Serilog.Sinks.File",
    "Serilog.Sinks.NewRelic"
  ],
  "MinimumLevel": {
    "Default": "Information",
    "Override": {
      "Microsoft": "Information",
      "Microsoft.Hosting.Lifetime": "Debug"
    }
  },
  "WriteTo": [
    {
      "Name": "File",
      "Args": {
        "formatter": "AC.Serilog.Formatting.Formatting.Compact.ExtendedCompactJsonFormatter, AC.Serilog.Formatting",
        "path": "F:\\Sales-Logs\\OrderBuilderApi\\OrderBuilder-.log",
        "fileSizeLimitBytes": 1073741824, //1GiB
        "retainedFileCountLimit": 5,
        "rollingInterval": "Day",
        "rollOnFileSizeLimit": false
      }
    },
    {
      "Name": "NewRelic",
      "Args": {
        "applicationName": "OrderBuilder"
      }
    }
  ]
}
```

This avoids the need to hardcode any Serilog sink configuration within the codebase itself.

File: src/Serilog.Sinks.NewRelic/Serilog.Sinks.NewRelic.nuspec (hunk 1 above)
Comment (stanisls) [reply src/Serilog.Sinks.NewRelic/Serilog.Sinks.NewRelic.nuspec:1]: The file structure is correct, so must be caching. And do add your name to the authors list.

**Comment by stanisls** [comment]
Thank for the contribution and extended answers. I'll do some local tests over the week-end before building a new nuget.

**Pull request merged**